src/MauiMicroMvvm/Internals/DefaultNavigation.cs

```diff
@@ -5,22 +5,23 @@ namespace MauiMicroMvvm.Internals;
 [EditorBrowsable(EditorBrowsableState.Never)]
 public class DefaultNavigation<TShell> : INavigation where TShell : Shell
 {
-    private readonly TShell _shell;
+    private readonly Lazy<TShell> _lazyShell;
+    private TShell Shell => _lazyShell.Value;
 
-    public DefaultNavigation(TShell shell)
+    public DefaultNavigation(IServiceProvider services)
```

Comment:
You should avoid injecting the IServiceProvider unless it is absolutely necessary. In this case it make more sense to simply add a registration for `Lazy<TShell>` and inject it directly.